lenskit-core/src/main/java/org/grouplens/lenskit/data/event/Like.java

```diff
@@ -20,14 +20,120 @@
  */
 package org.grouplens.lenskit.data.event;
 
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.io.Serializable;
+
 /**
  * Generic interface representing a unary interaction.  This provides a basic, generic representation
  * of likes, clicks, purchases, and other unary interactions.  Applications may want to implement
  * their own events to represent such interactions.
+ * <p>
+ * This class can be created with {@link LikeBuilder} or the {@link #create(long, long, long)} method.
+ * </p>
  *
  * @since 2.2
  * @see Events#like(long, long)
  * @see Events#like(long, long, long)
  */
-public interface Like extends Event {
+public final class Like implements Event, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final long userId;
+    private final long itemId;
+    private final long timestamp;
+
+    Like(long uid, long iid, long ts) {
```

Comment:
There are a variety of reasons to make constructors non-public. Many of them do not apply in this case (implementation flexbility, type inference for generic class construction, etc.).

One reason is consistency: if we are going to use the factory method and/or builder patterns for other, more complicated classes in LensKit, it seems to me that it will be easier to work with the code if things are generally done one way. 'LensKit classes are constructed with factory methods' (with the exception of components that participate in dependency injection) is a single rule, rather than having to determine for each class whether it is complicated enough that we decided to use factory methods or builders. And when we write a class, deciding if it will forever be simple enough that we can commit to a public constructor.

I could be dissuaded from this. It may be a good idea to leave the event constructors public, but use factory methods elsewhere.